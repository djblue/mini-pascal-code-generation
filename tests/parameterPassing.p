program main;

class main 
begin
  function square (aa : integer) : integer;
    var bb : integer;
  begin
    print aa;
    square := aa * aa
  end;
  function main;
    var res : integer;
  begin
    res := square(128);
    print res
  end
end
.
