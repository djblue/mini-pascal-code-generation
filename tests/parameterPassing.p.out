128
16384
0
1048576
